appdaemontestframework/hass_mocks.py

```diff
@@ -1,12 +1,32 @@
 import logging
 import mock
 from appdaemon.plugins.hass.hassapi import Hass
+import appdaemon.utils
+from packaging import version
 import textwrap
 import warnings
 
 
+class _DeprecatedAppdaemonWarning:
+    # Use class vars so we can keep sticky state to only warn once per test session
+    already_warned_deprecated_appdaemon_version = False
+    min_supported_version = version.Version('4.0.0')
+
+    def __init__(self):
+        if self.already_warned_deprecated_appdaemon_version:
+            return
+        self.already_warned_deprecated_appdaemon_version = True
+
+        appdaemon_version = version.Version(appdaemon.utils.__version__)
+        if appdaemon_version < self.min_supported_version :
```

Comment:
There was already a check for versions below '4.0.0' in `automation_fixture.py`. Let's unify them. 
Your way seems more reliable (I just checked on the string previously),  so I'm thinking of using that.

I have your code running and it's an easy refactor so I'll do it now.

Reply:
👍 go right ahead with that refactor

Reply:
Alright I'm done :)